### PR TITLE
Feature/multilayer view algorithms

### PR DESCRIPTION
**Pull request by ljeub-pometry**
Make use off and test multilayer views (WIP)

### CODE REVIEW

**Comment by ljeub-pometry** [comment]
- algorithms that call `multilayerView` can now make use of `ExplodedVertex` methods without using `asInstanceOf`
- `GraphFunction`s no longer have explicit `Vertex` type to allow passing along functions using `ExplodedVertex` or `Vertex`
- The fact that `RaphtoryGraph` directly exposes graph operations means that we require a lot of abstract types. 
- `RaphtoryGraph` also does not support multilayer views properly which means using `asInstanceOf` is necessary when using `step`/`iterate` from `RaphtoryGraph` directly

**Comment by ljeub-pometry** [comment]
Documentation needs to be updated for the new type parameters once this is settled

File: core/src/main/scala/com/raphtory/algorithms/temporal/community/MultilayerLPA.scala [hunk 1]
```diff
@@ -66,12 +66,12 @@ class MultilayerLPA(
     layerSize: Long,
     omega: Double = 1.0,
     seed: Long = -1
-) extends GraphAlgorithm {
+) extends Generic {
```
Comment (miratepuffin) [inline core/src/main/scala/com/raphtory/algorithms/temporal/community/MultilayerLPA.scala:69]: Can't wait to see this one properly rewritten!

File: core/src/main/scala/com/raphtory/api/algorithm/Multilayer.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          with Multilayer {
+
+    override def apply(graph: MultilayerGraphPerspective): graph.MultilayerGraph =
+      second(first(graph).clearMessages())
+
+    override def tabularise(graph: MultilayerGraphPerspective): Table =
+      second.tabularise(graph)
+  }
+
+  case class ChainedMultilayer2(
```
Comment (miratepuffin) [inline core/src/main/scala/com/raphtory/api/algorithm/Multilayer.scala:26]: I am not saying I have a better one, but we need to come up with a better name for this 😄

File: core/src/main/scala/com/raphtory/api/graphview/Deployment.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+/** Public interface for the Raphtory deployment
+  *
+  * @see [[DeployedTemporalGraph]]
+  */
+trait Deployment {
+
+  /** The configuration used by the deployment
+    */
+  val conf: Config
```
Comment (miratepuffin) [inline core/src/main/scala/com/raphtory/api/graphview/Deployment.scala:13]: Cheers for adding this!

File: core/src/main/scala/com/raphtory/api/graphview/RaphtoryGraph.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import com.raphtory.client.QuerySender
+import com.raphtory.components.querymanager.Query
+
+/** Reduced GraphView with fixed timeline
+  *
+  * This [[GraphView]] is returned by the [[DottedGraph]] operations and does not support further timeline manipulation.
+  *
+  * @see [[GraphView]], [[MultilayerRaphtoryGraph]], [[DottedGraph]]
+  */
+class RaphtoryGraph private[api] (
```
Comment (miratepuffin) [inline core/src/main/scala/com/raphtory/api/graphview/RaphtoryGraph.scala:12]: This is an odd way round to have the private no?

**Review by miratepuffin — APPROVED** [review]
As discussed we need to finish the docs (which are looking *chef's kiss*) but all good to go!

**Pull request merged**